File: JAVA/ws_rest/consuming-rest/src/main/java/Services/IUserService.java
package Services;

import Sockets.Models.User;

import java.io.IOException;

public interface IUserService {

    //Test for getting a receipt for a specific user
    User GetUser(String username)
            throws IOException, ClassNotFoundException;
    User ValidateLogin(String username, String password)
        throws IOException, ClassNotFoundException;

    User ValidateRegister(User user)
        throws IOException, ClassNotFoundException;

    User editUser(User user);

    User changePassword(User user);
}

File: C#/WebSiteSEP3/Client/Data/IUserService.cs
using System;
using System.Threading.Tasks;
using Client.Models;

namespace Client.Data
{
    public interface IUserService
    {
        Task<User> GetUser(string username);
        Task<User> ValidateLogin(string username, string password);
        
        Task<User> RegisterUserAsync(User user);
        Task<User> EditUser(User user);
        Task<User> ChangePassword(User user);
    }
}